src/clib/ct.cpp

```diff
@@ -350,9 +350,9 @@ extern "C" {
     int thermo_getMoleFractions(int n, size_t lenx, double* x)
     {
         try {
-            ThermoPhase& p = ThermoCabinet::item(n);
-            p.checkSpeciesArraySize(lenx);
-            p.getMoleFractions(x);
+            auto p = ThermoCabinet::at(n);
```

Comment:
Here and for all similar usage, I think you want to specify the type as `auto&`. This will avoid needing to copy the `shared_ptr` object and manipulate the reference count for a very short-lived object.

Reply:
:+1: Good point